content/en/blog/_posts/2025-02-21-introducing-gateway-api-inference-extension/index.md

```diff
@@ -0,0 +1,128 @@
+(GPU) utilization for AI workloads.
+
+## How It Works
+
+The design introduces two new Custom Resources (CRDs) with distinct responsibilities, each aligning with a
+specific user persona in the AI/ML serving workflow​:
+
+{{< figure src="inference-extension-resource-model.png" alt="Resource Model" class="diagram-large" clicktozoom="true" >}}
+
+1. [InferencePool](https://gateway-api-inference-extension.sigs.k8s.io/api-types/inferencepool/)
+   Defines a pool of pods (model servers) running on shared compute (e.g., GPU nodes). The platform admin can
+   configure how these pods are deployed, scaled, and balanced. An InferencePool ensures consistent resource
+   usage and enforces platform-wide policies. An InferencePool is similar to a Service but specialized for AI/ML
+   serving needs and aware of the model-serving protocol.
+
+2. [InferenceModel](https://gateway-api-inference-extension.sigs.k8s.io/api-types/inferencemodel/)
+   A user-facing model endpoint managed by AI/ML owners. It maps a public name (e.g., "gpt-4-chat") to the actual
+   model within an InferencePool. This lets workload owners specify which models (and optional fine-tuning) they
+   want served, plus a traffic-splitting or prioritization policy.
+
+In summary, the InferenceModel API lets AI/ML owners manage what is served, while the InferencePool lets platform
+operators manage where and how it’s served.
+
+## Request Flow
+
+The flow of a request builds on the Gateway API model (Gateways and HTTPRoutes) with one or more extra inference-aware
+steps, e.g. extensions, in the middle. Here’s a high-level example of the request flow with the
+[​Endpoint Selection Extension](https://gateway-api-inference-extension.sigs.k8s.io/#endpoint-selection-extension) (ESE):
+
+{{< figure src="inference-extension-request-flow.png" alt="Request Flow" class="diagram-large" clicktozoom="true" >}}
+
+1. **Gateway Routing**
+   A client sends a request (e.g., an HTTP POST to /completions). The Gateway (like Envoy) examines the HTTPRoute
+   and identifies the matching InferencePool backend.
+
+2. **Endpoint Selection**
+   Instead of simply forwarding to any pod, the Gateway consults an inference-specific routing extension, e.g. ​endpoint selection extension. This
+   extension examines live pod metrics—queue lengths, memory usage, loaded adapters—to pick the best pod for the
+   request.
+
+3. **Inference-Aware Scheduling**
+   The chosen pod is the one that can handle the request with the lowest latency or highest efficiency, given the
+   user’s criticality or resource needs. The Gateway then forwards traffic to that specific pod.
+
+{{< figure src="inference-extension-epp-scheduling.png" alt="Endpoint Extension Scheduling" class="diagram-large" clicktozoom="true" >}}
+
+This extra step provides a smarter, model-aware routing mechanism that still feels like a normal single request to
+the client.
+
```

Comment:
Somewhere in this section I think it would be useful to mention the extensible nature of this model, and that new extensions can be developed that will be compatible with any Inference Gateway.

Reply:
I updated this section based on ^ feedback, PTAL.